src/lib/utils/sqlite_wrapper.hpp

```diff
@@ -31,33 +30,50 @@ class SQLiteWrapper final {
    * @param table      The table to load into sqlite
    * @param tablename  The desired table name
    */
-  void create_table(const Table& table, const std::string& table_name);
+  void create_sqlite_table(const Table& table, const std::string& table_name);
 
-  /*
-   * Executes a sql query in the sqlite database context and returns a Hyrise table.
-   *
-   * @param sql_query Query to be executed
-   * @returns An opossum Table containing the results of the executed query
-   */
-  std::shared_ptr<Table> execute_query(const std::string& sql);
+  class Connection final {
+   public:
+    explicit Connection(const std::string& uri);
+
+    Connection(const Connection&) = delete;
+    Connection(Connection&&) noexcept;
+    Connection& operator=(const Connection&) = delete;
+    Connection& operator=(Connection&&) = delete;
+    ~Connection();
+
+    sqlite3* db{nullptr};
+
+    /*
+     * Executes a sql query in the sqlite database context and returns a Hyrise table.
```

Comment:
I'm not sure how `sql query` is defined. Does it comprise 1..n statements by definition? If not, "Executes all sql queries of the given sql string..." might fit better.

Reply:
I am not even sure if SQL allows for multiple queries. Changed it to "Executes a sql string with potentially multiple statements"

Reply:
Ok, good!